test/parallel/test-crypto.js

```diff
@@ -161,45 +161,41 @@ testImmutability(tls.getCiphers);
 testImmutability(crypto.getHashes);
 testImmutability(crypto.getCurves);
 
-const encodingError = {
-  code: 'ERR_INVALID_ARG_VALUE',
-  name: 'TypeError',
-  message: "The argument 'encoding' is invalid for data of length 1." +
-           " Received 'hex'",
-};
-
 // Regression tests for https://github.com/nodejs/node-v0.x-archive/pull/5725:
 // hex input that's not a power of two should throw, not assert in C++ land.
-['createCipher', 'createDecipher'].forEach((funcName) => {
-  assert.throws(
-    () => crypto[funcName]('aes192', 'test').update('0', 'hex'),
-    (error) => {
-      assert.ok(!('opensslErrorStack' in error));
-      if (common.hasFipsCrypto) {
-        return error instanceof Error &&
-               error.name === 'Error' &&
-               /^Error: not supported in FIPS mode$/.test(error);
-      }
-      assert.throws(() => { throw error; }, encodingError);
-      return true;
-    }
-  );
+assert.throws(function() {
+  crypto.createCipher('aes192', 'test').update('0', 'hex');
+}, (err) => {
+  const errorMessage =
+    common.hasFipsCrypto ? /not supported in FIPS mode/ : /Bad input string/;
+  // Throws general Error, so there is no opensslErrorStack property.
+  if ((err instanceof Error) &&
+      errorMessage.test(err) &&
+      err.opensslErrorStack === undefined) {
+    return true;
+  }
 });
 
-assert.throws(
-  () => crypto.createHash('sha1').update('0', 'hex'),
-  (error) => {
-    assert.ok(!('opensslErrorStack' in error));
-    assert.throws(() => { throw error; }, encodingError);
+assert.throws(function() {
+  crypto.createDecipher('aes192', 'test').update('0', 'hex');
+}, (err) => {
+  const errorMessage =
+    common.hasFipsCrypto ? /not supported in FIPS mode/ : /Bad input string/;
+  // Throws general Error, so there is no opensslErrorStack property.
+  if ((err instanceof Error) &&
+      errorMessage.test(err) &&
+      err.opensslErrorStack === undefined) {
     return true;
   }
 });
 
-assert.throws(
-  () => crypto.createHmac('sha256', 'a secret').update('0', 'hex'),
-  (error) => {
-    assert.ok(!('opensslErrorStack' in error));
-    assert.throws(() => { throw error; }, encodingError);
```

Comment:
I suggest to keep the old change and just replace `encodingError` with `/^TypeError: Bad input string$/`. Similar with the entries above. That way most lines are backported, while keeping the old behavior.

Reply:
Makes sense